src/values/AbstractValue.js

```diff
@@ -436,7 +359,10 @@ export default class AbstractValue extends Value {
     }
 
     let resultTypes = TypesDomain.binaryOp(op, leftTypes, rightTypes);
-    let resultValues = ValuesDomain.binaryOp(realm, op, leftValues, rightValues);
+    let resultValues =
```

Comment:
Why is this change part of an other behavior preserving pull request?

Reply:
The pull request is not entirely behavior preserving because a few simplifications done inside the factories are now replaced by full scale simplification. One of the effects of that is that templates now end up being asked for their truthiness and this breaks things. It is basically a bug that has been uncovered and it is convenient to fix it here.